src/consensus/tx_verify.cpp

```diff
@@ -298,3 +298,89 @@ bool Consensus::CheckTxInputs(const CTransaction &tx, CValidationState &state, c
         return state.DoS(100, false, REJECT_INVALID, "bad-txns-fee-outofrange");
     return true;
 }
+
+bool Consensus::CheckTxInputs(const CTransaction &tx,
+    CValidationState &state,
+    const CCoinsViewCache &inputs,
+    std::map<COutPoint, Coin> *blockInputs)
+{
+    // This doesn't trigger the DoS code on purpose; if it did, it would make it easier
+    // for an attacker to attempt to split the network.
+    assert(blockInputs != nullptr);
+    if (!tx.IsCoinBase())
+    {
+        for (unsigned int i = 0; i < tx.vin.size(); i++)
+        {
+            if (!inputs.HaveCoin(tx.vin[i].prevout) && !blockInputs->count(tx.vin[i].prevout))
+            {
+                return state.Invalid(false, 0, "", "Inputs unavailable");
+            }
+        }
+    }
```

Comment:
Seems like a lot of duplicated code in this second CheckTxInputs()...the only difference I can see, unless mistaken, is this top section for checkout when blockInputs is not null.  I think you could easily make a switch there if the pointer is not null to check using that method and revert to the old standard way, if blockIputs is NULL?

Reply:
only differences are where it might use the blockinputs to find an input vs the coinsviewcache, I suppose it would be easier to maintain if i just added nullptr checks inside instead of making an overload. thanks for feedback